### PR TITLE
added more security user providers

### PR DESCRIPTION
**Pull request by ro0NL**

### CODE REVIEW

File: src/User/Infra/Security/SecurityUser.php [hunk 1]
```diff
@@ -27,11 +27,7 @@ public function __construct(User $user, array $roles = [])
         $credential = $user->getCredential();
         if ($credential instanceof PasswordProtectedInterface) {
             $this->password = $credential->getPassword();
-
-            $algorithm = $credential->getPasswordAlgorithm();
-            if (null !== $algorithm->salt) {
-                $this->passwordSalt = $algorithm->salt->token;
-            }
+            $this->passwordSalt = null === ($salt = $credential->getPasswordAlgorithm()->salt) ? null : $salt->token;
```
Comment (azjezz) [inline src/User/Infra/Security/SecurityUser.php]: you can use Null coalescing operator here : 

`$this->passwordSalt = $credential->getPasswordAlgorithm()->salt ?? null;`

File: src/User/Infra/Security/Oauth/EavSecurityUserProvider.php [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+use HWI\Bundle\OAuthBundle\OAuth\ResourceOwnerInterface;
+use HWI\Bundle\OAuthBundle\OAuth\Response\UserResponseInterface;
+use HWI\Bundle\OAuthBundle\Security\Core\User\OAuthAwareUserProviderInterface;
+use MsgPhp\Domain\Exception\EntityNotFoundException;
+use MsgPhp\Domain\Factory\EntityAwareFactoryInterface;
+use MsgPhp\Domain\Message\{DomainMessageBusInterface, MessageDispatchingTrait};
+use MsgPhp\Eav\Entity\Attribute;
+use MsgPhp\User\Command\{AddUserAttributeValueCommand, ConfirmUserCommand, CreateUserCommand};
+use MsgPhp\User\Entity\{User, UserAttributeValue};
+use MsgPhp\User\Infra\Security\SecurityUserProvider as BaseSecurityUserProvider;
```
Comment (azjezz) [inline src/User/Infra/Security/Oauth/EavSecurityUserProvider.php]: there's no need for an alias here

**Review by azjezz — CHANGES_REQUESTED** [review]

File: src/User/Infra/Security/SecurityUser.php (hunk 1 above)
Comment (ro0NL) [reply src/User/Infra/Security/SecurityUser.php]: but we need $salt->token :)

File: src/User/Infra/Security/Oauth/EavSecurityUserProvider.php (hunk 2 above)
Comment (ro0NL) [reply src/User/Infra/Security/Oauth/EavSecurityUserProvider.php]: `SecurityUserProvider` already exists in the current namespace, but we want to inject `SecurityUserProvider` from the parent namespace.

but im leaning to move the entire Oauth namespace to Infra/HWIOAuth

**Comment by ro0NL** [comment]
Hi @azjezz thx for having a look :) the PR got a bit outdated over time. But it's still on my list.

However as said, i think we need to move things to `Infra\HWIOAuth`

File: src/User/Infra/Security/SecurityUser.php (hunk 1 above)
Comment (azjezz) [reply src/User/Infra/Security/SecurityUser.php]: oh my fault i meant : 

```php
$this->passwordSalt = $credential->getPasswordAlgorithm()->salt->token ?? null;
```
and yes, it works, even if salt is null : https://3v4l.org/PTRrr

File: src/User/Infra/Security/Oauth/EavSecurityUserProvider.php (hunk 2 above)
Comment (azjezz) [reply src/User/Infra/Security/Oauth/EavSecurityUserProvider.php]: if are not using the `SecurityUserProvider` in the current namespace you don't have to use an alias.

File: src/User/Infra/Security/SecurityUser.php (hunk 1 above)
Comment (azjezz) [reply src/User/Infra/Security/SecurityUser.php]: done. #216

**Pull request closed without merging**